LatLonShapeDocValuesQueryTests.testIndexMultiShapes

### CI Link

n/a

### Repro line

./gradlew ':x-pack:plugin:spatial:test' --tests "org.elasticsearch.xpack.spatial.index.mapper.LatLonShapeDocValuesQueryTests.testIndexMultiShapes" -Dtests.seed=368A99A4FD18272B -Dtests.locale=lv -Dtests.timezone=Africa/Niamey -Druntime.java=17

### Does it reproduce?

Yes

### Applicable branches

7.17

### Failure history

_No response_

### Failure excerpt

```
java.lang.AssertionError: expected:<0> but was:<1>
        at __randomizedtesting.SeedInfo.seed([368A99A4FD18272B:EABCA164BFB771B4]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:645)
        at org.junit.Assert.assertEquals(Assert.java:631)
        at org.elasticsearch.xpack.spatial.index.mapper.LatLonShapeDocValuesQueryTests.assertQueries(LatLonShapeDocValuesQueryTests.java:148)
        at org.elasticsearch.xpack.spatial.index.mapper.LatLonShapeDocValuesQueryTests.testIndexMultiShapes(LatLonShapeDocValuesQueryTests.java:141)
```

--- a/x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/mapper/LatLonShapeDocValuesQueryTests.java
+++ b/x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/mapper/LatLonShapeDocValuesQueryTests.java
@@ -136,6 +136,10 @@ public class LatLonShapeDocValuesQueryTests extends ESTestCase {
         for (int i = 0; i < 25; i++) {
             LatLonGeometry[] geometries = randomLuceneQueryGeometries();
             for (ShapeField.QueryRelation relation : ShapeField.QueryRelation.values()) {
+                if (relation == ShapeField.QueryRelation.CONTAINS) {
+                    // We don't check this here as it might fail due to LUCENE-10514
+                    continue;
+                }
                 Query indexQuery = LatLonShape.newGeometryQuery(FIELD_NAME, relation, geometries);
                 Query docValQuery = new LatLonShapeDocValuesQuery(FIELD_NAME, relation, geometries);
                 assertQueries(s, indexQuery, docValQuery, numDocs);

